src/extension.ts

```diff
@@ -115,11 +173,35 @@ export function activate(): void {
         }
         sourcegraph.internal.updateContext(context)
     }
-    sourcegraph.configuration.subscribe(() => updateContext())
-    sourcegraph.workspace.onDidOpenTextDocument.subscribe(() => updateContext())
-    if (sourcegraph.workspace.onDidChangeRoots) {
-        sourcegraph.workspace.onDidChangeRoots.subscribe(() => updateContext())
-    }
+
+    // Update the context when the configuration, workspace roots or active editors change.
+    context.subscriptions.add(
+        combineLatest([
+            configurationChanges.pipe(
+                map(settings => settings['codecov.endpoints'])
+            ),
+            from(
+                // Backcompat: rely on onDidChangeRoots if the extension host doesn't support rootChanges.
+                sourcegraph.workspace.rootChanges ||
+                    sourcegraph.workspace.onDidChangeRoots
+            ).pipe(
+                map(() => sourcegraph.workspace.roots),
+                startWith(sourcegraph.workspace.roots)
+            ),
+            editorsChanges,
+            // tslint:disable-next-line: rxjs-no-async-subscribe
+        ])
+            .pipe(
+                concatMap(([endpoints, roots, editors]) =>
+                    updateContext(endpoints, roots, editors)
+                ),
+                catchError(err => {
```

Comment:
I think this `catchError` is too late, it won't resubscribe the Observable. The catch would need to happen within the `concatMap` (could be an async fn with `try`/`catch`)